### PR TITLE
fix: update generate default for schema

### PR DESCRIPTION
**Pull request by aiwantaozi**
**Problem:**
 Now will generate below computed attributes cause terraform validate failed.
```json
{
   "seeding": {
       "type": "none",
        "url": {
            "storage": {
                "size": 10240
             }
        }
   }
}
```
**Solution:**
Update generate computed attributes strategy, only generate optional(fieldname, "default") while the object default existed.
After updated:
```json
{
   "seeding": {
       "type": "none"
   }
}
```

**Related Issue:**
#1900

### CODE REVIEW

File: pkg/templates/openapi/default.go [hunk 1]
```diff
@@ -42,41 +44,80 @@ func GenSchemaDefaultPatch(ctx context.Context, schema *openapi3.Schema) ([]byte
...
+				var b []byte
+
+				p, err := sjson.SetBytes(b, nodes[0].GetID(), def)
+				if err != nil {
+					resultChan <- patchResult{
+						group: name,
+						err:   fmt.Errorf("set patch node error for %s %s: %w", nodes[0].GetID(), string(p), err),
+					}
+				}
+
```
Comment (thxCode) [inline pkg/templates/openapi/default.go:66]: can we construct a path bytes without using sjson? if no, here also need to return inside the `err != nil` block.

File: pkg/templates/openapi/default.go [hunk 2]
```diff
@@ -42,41 +44,80 @@ func GenSchemaDefaultPatch(ctx context.Context, schema *openapi3.Schema) ([]byte
...
+				// Generate root patch.
+				var b []byte
+
+				p, err := sjson.SetBytes(b, nodes[0].GetID(), nodes[0].GetDefault())
+				if err != nil {
+					resultChan <- patchResult{
+						group: name,
+						err:   fmt.Errorf("set patch node error for %s %s: %w", nodes[0].GetID(), string(p), err),
+					}
+				}
```
Comment (thxCode) [inline pkg/templates/openapi/default.go:90]: ditto

File: pkg/templates/openapi/stack.go [hunk 3]
```diff
@@ -86,35 +89,29 @@ func (n *DefaultPatchNode) Children() []Node {
...
-				log.Errorf("failed to set default value for %s: %v", p, err)
-			} else {
-				sdn.patch = patch
+		var (
+			p   = n.GetID() + ".0"
+			sdn = &DefaultPatchNode{
+				Schema: n.Schema.Items.Value,
+				id:     p,
 			}
+		)
```
Comment (thxCode) [inline pkg/templates/openapi/stack.go:102]: sdn := &DefaultPatchNode{
  Schema: n.Schema.Items.Value

  id : n.GetID() + ".0",
  def: n.Default
}

return []Node{sdn}

File: pkg/templates/openapi/stack.go [hunk 4]
```diff
@@ -70,14 +80,7 @@ func (n *DefaultPatchNode) Children() []Node {
...
-				patch, err := sjson.SetBytes(b, p, prop.Value.Default)
-				if err != nil {
-					log.Errorf("failed to set default value for %s: %v", p, err)
-				} else {
-					sdn.patch = patch
-				}
+				sdn.def = prop.Value.Default
 			}
 
 			children = append(
```
Comment (thxCode) [inline pkg/templates/openapi/stack.go:86]: preallocate `children` slice.

File: pkg/templates/openapi/default.go [hunk 5]
```diff
@@ -42,41 +44,80 @@ func GenSchemaDefaultPatch(ctx context.Context, schema *openapi3.Schema) ([]byte
...
+						}
+					}
+
+					var (
+						pid = np.GetParentID()
+						pr  = gjson.ParseBytes(p)
+					)
+
+					if pe := pr.Get(pid); pe.Exists() {
+						p, err = jsonpatch.MergePatch(npb, p)
```
Comment (thxCode) [inline pkg/templates/openapi/default.go:115]: if an error occurs, the p will be reset, we should allocate a new local var to receive the merged result, and then assign it to p if no error occurs.

File: pkg/templates/openapi/default.go [hunk 6]
```diff
@@ -42,41 +44,80 @@ func GenSchemaDefaultPatch(ctx context.Context, schema *openapi3.Schema) ([]byte
...
+				if def == nil {
 					resultChan <- patchResult{
 						group: name,
 					}
 
 					return
 				}
 
+				// Generate patch.
+				var b []byte
```
Comment (thxCode) [inline pkg/templates/openapi/default.go:57]: remove this declaration

File: pkg/templates/openapi/default.go [hunk 7]
```diff
@@ -42,41 +44,80 @@ func GenSchemaDefaultPatch(ctx context.Context, schema *openapi3.Schema) ([]byte
...
+						patch: nil,
 					}
 
-					if len(p) == 0 {
-						p = np
+					return
+				}
+
+				// Generate root patch.
+				var b []byte
```
Comment (thxCode) [inline pkg/templates/openapi/default.go:82]: ditto

File: pkg/templates/openapi/default.go [hunk 8]
```diff
@@ -42,41 +44,80 @@ func GenSchemaDefaultPatch(ctx context.Context, schema *openapi3.Schema) ([]byte
...
+
+				for i := 0; i < len(nodes); i++ {
+					np := nodes[i]
+					if np.GetDefault() == nil {
 						continue
 					}
 
-					p, err = jsonpatch.MergePatch(p, np)
+					// Generate sub node patch.
+					var nb []byte
```
Comment (thxCode) [inline pkg/templates/openapi/default.go:99]: remove this declaration.

File: pkg/templates/openapi/default.go (hunk 5 above)
Comment (aiwantaozi) [reply pkg/templates/openapi/default.go:124]: Update to return error while error occurs

**Review by thxCode — APPROVED** [review]
LGTM

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**